### ISSUE TITLE
Analyze prebuilts for ASPNET Extensions

### ISSUE THREAD

**Issue by dseefeld** [dotnet/source-build#1132]
The following prebuilts are added by the ASPNET Extensions repo:

These look like they could be reference-only packages.  Listed in the table below are the versions of the packages that are already being generated as reference-only.  Is4.6.0-preview7.19352.8 the required version for each of these packages?

PackageName | PackageVersion | Ref-Pkg Versions | Category
-- | -- | -- | --
System.ComponentModel.Annotations   | 4.6.0-preview7.19352.8 |  |
System.Diagnostics.DiagnosticSource |   4.6.0-preview7.19352.8 | 4.0.0, 4.3.0, 4.4.0 |
System.Diagnostics.EventLog |   4.6.0-preview7.19352.8 |  |
System.Runtime.CompilerServices.Unsafe |   4.6.0-preview7.19352.8 |  |
System.Security.Cryptography.Xml |   4.6.0-preview7.19352.8 |  |
System.Text.Encodings.Web |   4.6.0-preview7.19352.8 |  |
System.Text.Json | 4.6.0-preview7.19352.8  |  |

**Comment by JunTaoLuo** [comment]
TL;DR, we need System.Diagnostics.EventLog and System.Security.Cryptography.Xml from CoreFX, we can stop building System.Diagnostics.DiagnosticSource. The rest of the Extensions packages need to compile for netstandard2.1 and remove dependencies.

I've looked into these dependencies and here's my recommendations:

PackageName | PackageVersion | Ref-Pkg Versions | Category | Referenced By | Recommendation
-- | -- | -- | -- | -- | --
System.ComponentModel.Annotations | 4.6.0-preview6.19303.8 |   |   |  Microsoft.Extensions.Options.DataAnnotations, Microsoft.Extensions.Options | Extension packages are included in shared framework, as is the CoreFX package. Cross compile for netstandard2.1 and remove dependency https://github.com/aspnet/Extensions/issues/2056
System.Diagnostics.DiagnosticSource | 4.6.0-preview6.19303.8 | 4.0.0, 4.3.0, 4.4.0 |  | Microsoft.Extensions.DiagnosticAdapter | Extension package is not included in the shared framework. Skip building it in source build
System.Diagnostics.EventLog | 4.6.0-preview6.19303.8 |  |  | Microsoft.Extensions.Logging.EventLog | Extensions package is included in shared framework but CoreFx package is not. CoreFx needs to build this package.
System.Memory | 4.5.2 | 4.5.2 | Ref-Only | Microsoft.Extensions.Primitives | Extension package is included in shared framework, as is the CoreFX package. Cross compile for netstandard2.1 and remove dependency https://github.com/aspnet/Extensions/issues/2056
System.Runtime.CompilerServices.Unsafe | 4.6.0-preview6.19303.8 |   |  | Microsoft.Extensions.Primitives | Extension package is included in shared framework, as is the CoreFX package. Cross compile for netstandard2.1 and remove dependency https://github.com/aspnet/Extensions/issues/2056
System.Security.Cryptography.Xml | 4.6.0-preview6.19303.8 |   |  | Microsoft.Extensions.Configuration.Xml | Extensions package is included in shared framework but CoreFx package is not. CoreFx needs to build this package.
System.Text.Encodings.Web | 4.6.0-preview6.19303.8 |   |  | Microsoft.Extensions.WebEncoders | Extension package is included in shared framework, as is the CoreFX package. Cross compile for netstandard2.1 and remove dependency https://github.com/aspnet/Extensions/issues/2056
System.Text.Json | 4.6.0-preview6.19303.8 |   |  | Microsoft.Extensions.Configuration.Json, Microsoft.JSInterop, Microsoft.Extensions.Logging.EventSource | Extension packages are included in shared framework, as is the CoreFX package. Cross compile for netstandard2.1 and remove dependency https://github.com/aspnet/Extensions/issues/2056
System.Threading.Tasks.Extensions | 4.5.2 | 4.0.0, 4.3.0, 4.5.1 | |  Microsoft.Extensions.Configuration.Json, Microsoft.Extensions.Logging.EventSource | Extension packages are included in shared framework, as is the CoreFX package. Cross compile for netstandard2.1 and remove dependency https://github.com/aspnet/Extensions/issues/2056

**Comment by JunTaoLuo** [comment]
Is there another list for AspNetCore-Tooling?

**Comment by JunTaoLuo** [comment]
FYI, the recommendation has been updated to cross compile for netcoreapp3.0 instead of netstandard2.1

**Comment by JunTaoLuo** [comment]
@dseefeld Can this issue be closed now? We have addressed the underlying issues already. Or is this being kept open to track when the fixes make their way to source build?

**Comment by dseefeld** [comment]
We usually keep them open until our prebuilt report shows no outstanding prebuilts.

**Comment by JunTaoLuo** [comment]
I think this is completed @dseefeld ?

**Comment by dseefeld** [comment]
Yes.  All prebuilts have been removed.
